src/test/java/com/iexec/core/sms/SmsServiceTests.java

```diff
@@ -20,12 +24,17 @@
     @Mock
     private SmsClient smsClient;
 
-    @InjectMocks
+    @Mock
+    private SmsConfiguration configuration;
+
     private SmsService smsService;
 
     @BeforeEach
     void init() {
         MockitoAnnotations.openMocks(this);
+        when(configuration.getSmsURL()).thenReturn("http://localhost");
+        smsService = new SmsService(configuration);
+        ReflectionTestUtils.setField(smsService, "smsClient", smsClient);
```

Comment:
To avoid reflection, what about something like:
```
    @Mock
    private SmsClientService smsClientService;

    @InjectMocks
    private SmsService smsService;

    @BeforeEach
    void init() {
        MockitoAnnotations.openMocks(this);
        when(smsClientService.getClient()).thenReturn(mock(SmsClient.class));
    }
```
```
@Service
public class SmsClientService {
    private final SmsConfiguration configuration;

    public SmsClientService(SmsConfiguration configuration) {
        this.configuration = configuration;
    }

    public SmsClient getClient() {
        return SmsClientBuilder.getInstance(Logger.Level.BASIC, configuration.getSmsURL());
    }
}
```
?